### PR TITLE
attributes: 'eq' and 'repr' methods for Name attr.

### PR DESCRIPTION
**Pull request by viktorTarasov**
_eq_, _repr_ and _str_ methods for _Name_ and it's inner _NameValue_ and _NameType_ classes
It will facilitate the coming implementation of _locate_ in _mem-repo_

### CODE REVIEW

File: kmip/core/attributes.py [hunk 1]
```diff
@@ -63,12 +63,35 @@ class NameValue(TextString):
         def __init__(self, value=None):
             super(Name.NameValue, self).__init__(value, Tags.NAME_VALUE)
 
+        def __eq__(self, other):
+            if isinstance(other, Name.NameValue):
+                if self.value == other.value:
+                    return True
+            return False
```
Comment (rellerreller) [inline kmip/core/attributes.py]: The other instances where __eq__ is implemented the final return statement returns NotImplemented for some reason. You would need to ask Peter why that is.

**Comment by rellerreller** [comment]
:-1: 

There should be some unit tests to verify the implementation.

**Comment by viktorTarasov** [comment]
fixing remarques from @rellerreller :
- _is-not-instance-of_ in '__eq__' returns _NotImplemented_;
- added unit tests for '__eq__' and '__str__' of _Name_, _NameType_ and _NameValue_.

**Comment by PeterHamilton** [comment]
@viktorTarasov This patch looks good overall. I'll give it another pass on the next update.

**Comment by viktorTarasov** [comment]
resolving comments of  @PeterHamilton:
- simplify return of _str_ magic method;
- no dummy test class to test _NotImplemented_ return.

**Comment by viktorTarasov** [comment]
Sorry,
re-fixed remarques of @PeterHamilton:
-  magic _str_ of _Name_ attribute returns just a string;
-  unit test adjusted accordingly.

**Comment by viktorTarasov** [comment]
rebased on actual _master_

**Pull request merged**